elks/arch/i86/boot/setup.S

```diff
@@ -262,6 +262,16 @@ no_blob:
 // Entry point for EMU86 ROM emulation
 
 start_os:
+#ifdef CONFIG_ARCH_SWAN
+        // Force color mode
+	mov	$0x80,%al
+	out	%al,$0x60
+
+	// Enable UART serial
+	mov	$0xE0,%al
+	out	%al,$0xB3
+#endif
```

Comment:
I would prefer to move this ifdef code into the arch-specific arch_set_initseg, but I can see that you can't because the hello message is displayed prior to the call. So we'll leave this hear for now until I think of whether to move the hello message there also or some other solution. It seems like making a call to arch_set_initseg immediately after clearing the setup data segment would be best for handling arch-specific system initialization.

Reply:
I can move *one* of them there, as `enabling color mode` (to enable 64 KiB of RAM, as opposed to 16) is only required later in the initialization process.